app/Helpers/database/achievement.php

```diff
@@ -65,9 +65,9 @@ function getAchievementsList(
         $query .= "AND pa.unlocked_at IS NULL ";
     }
 
-    if (isValidUsername($developer)) {
-        $bindings['author'] = $developer;
-        $query .= "AND ach.Author = :author ";
+    if (isValidUsername($developer?->username)) {
+        $bindings['author'] = $developer->username;
+        $query .= "AND ua.User = :author ";
```

Comment:
The point of this ticket is to move away from matching the string. Recommend:
```
$query .= "AND ach.user_id = $developer->id";
```

Reply:
Updated in latest to:

```php
$bindings['userId'] = $developer->id;
$query .= "AND ach.user_id = :userId ";
```